### PR TITLE
👌 IMPROVE: Legacy tar file archive migration performance

### PR DESCRIPTION
**Pull request by chrisjsewell**
Tar files do not allow performant single file streaming,
and so the file is fully extracted to disk,
before streaming to the new (zip file) archive.

### CODE REVIEW

**Review by unkcpz — CHANGES_REQUESTED** [review]
@chrisjsewell Thanks for the quick fix. I make the test on the tar achieve and it works great! Just a minor request from my side. And for the `base_parts` change, I have to admit I am not familiar with this part, so maybe @ramirezfranciscof can have another eye on this?

File: aiida/tools/archive/implementations/sqlite/migrations/legacy_to_new.py [hunk 1]
```diff
@@ -77,26 +79,36 @@ def perform_v1_migration(  # pylint: disable=too-many-locals
...
     ) as new_path:
-        with in_archive as path:
+        with in_archive_context(inpath) as path:
             length = sum(1 for _ in path.glob('**/*'))
+            base_parts = len(path.parts)
             with get_progress_reporter()(desc='Converting repo', total=length) as progress:
                 for subpath in path.glob('**/*'):
                     progress.update()
-                    parts = subpath.parts
+                    parts = subpath.parts[base_parts:]
```
[the remark this answers is not in the document]
Comment (chrisjsewell) [reply aiida/tools/archive/implementations/sqlite/migrations/legacy_to_new.py:111]: If we are inside an archive, then the path is always relative to the archive (`base_parts = 0`), but if we are inside a file system then the path may be absolute, and we want to get back to the relative path.

**Comment by chrisjsewell** [comment]
Ok, as it works I'm gonna merge 😄

**Review by unkcpz — APPROVED** [review]
Cool! Thanks @chrisjsewell

**Comment by chrisjsewell** [comment]
Cheers!

**Pull request merged**